gulp-tasks/transpile-typescript.js

```diff
@@ -0,0 +1,173 @@
+/*
+  Copyright 2019 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+const fs = require('fs-extra');
+const gulp = require('gulp');
+const ol = require('common-tags').oneLine;
+const path = require('path');
+const {rollup} = require('rollup');
+const resolve = require('rollup-plugin-node-resolve');
+const typescript2 = require('rollup-plugin-typescript2');
+const packageRunnner = require('./utils/package-runner');
+const logHelper = require('../infra/utils/log-helper');
+const {AsyncDebounce} = require('../infra/utils/AsyncDebounce');
+
+
+/**
+ * @type {string}
+ */
+const packagesDir = path.resolve('packages');
+
+/**
+ * Returns a Rollup plugin that replaces extensions within filenames and
+ * imports. This is a bit of a hack, but it's needed until to solve:
+ * https://github.com/rollup/rollup/issues/2847
+ *
+ * @param {string} oldExt
+ * @param {string} newExt
+ */
+const renameImportExtensionsPlugin = (oldExt, newExt) => {
+  return {
+    generateBundle: (options, bundle) => {
+      const importPattern =
+          new RegExp(`((?:import|export)[^']+')([^']+?)\\${oldExt}';`, 'g');
```

Comment:
Should we include `"` along with `'` as the possible quote characters?

Reply:
As long as we stick to our style guide, we shouldn't ever have double quotes. And since we're only matching our files, I think this is fine for now. Though I suppose if we add any third-party libraries (like maybe `idb`) then yeah, we'd have to change this.